Project/Project/main.asm

```diff
@@ -0,0 +1,158 @@
+;
+; Project.asm
+;
+; Created: 27/11/2018 19.25.32
+; Author : Yasin Issa Aden, Jaser Ghasemi, Modaser Ghasemi. 
+;
+
+
+; Replace with your application code
+
+.org 0x0000
+    jmp main
+
+; DDR* are used just for specifing what will be an input
+; pin, and what will be an output pin.
+; sbi stands for set bit in i/o register, which means we set 1 bit here.
+main:
+    sbi DDRB, 7 ;maps over digital pin nr. 13
+    sbi DDRB, 6 ;maps over digital pin nr. 12
+	sbi DDRB, 5 ;maps over digital pin nr. 11
+	sbi DDRB, 4 ;maps over digital pin nr. 10
+
+    sbi DDRE, 3 ;maps over digital pin nr. 5
+	sbi DDRG, 5 ;maps over digital pin nr. 4
+	sbi DDRE, 5 ;maps over digital pin nr. 3
+    
+   
+
+    cbi DDRE, 4  ; maps over digital ping nr 2
+    ; just set register r20 to 0
+    clr r20
+
+; loops polling the pin information to see when either
+; there's voltage (bit set - check_press_loop) on digital pin 2,
+; or not (bit clear - check_release_loop)
+check_press_loop:
+    ; if input on pin 2, skip next instruction
+    sbis PINE, 4
+    rjmp check_press_loop  ; this is the one we will skip
+    rjmp toggle_leds
+check_release_loop:
+    ; if input on pin 2, loop once more
+    ; if clear pin 2, switch to button press check loop
+    sbic PINE, 4
+    rjmp check_release_loop
+    rjmp check_press_loop
+
+toggle_leds:

```

Comment:
I think you might be able to reduce the complexity of your project a lot by using a single general purpose register to hold the state of the leds and then use the logic shift operations `lsl` and `lsr` to count up and down.